server/src/main/java/com/hedera/block/server/BlockStreamService.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Hedera Block Node
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server;
+
+import com.google.protobuf.Descriptors;
+import com.hedera.block.protos.BlockStreamServiceGrpcProto;
+import com.hedera.block.server.producer.ProducerBlockStreamObserver;
+import com.hedera.block.server.mediator.StreamMediator;
+import com.hedera.block.server.consumer.LiveStreamObserverImpl;
+import com.hedera.block.server.consumer.LiveStreamObserver;
+import io.grpc.stub.StreamObserver;
+import io.helidon.webserver.grpc.GrpcService;
+
+import java.util.logging.Logger;
+
+import static com.hedera.block.server.Constants.*;
+
+/**
+ * This class implements the GrpcService interface and provides the functionality for the BlockStreamService.
+ * It sets up the bidirectional streaming methods for the service and handles the routing for these methods.
+ * It also initializes the StreamMediator, BlockStorage, and BlockCache upon creation.
+ *
+ * <p>The class provides two main methods, streamSink and streamSource, which handle the client and server streaming
+ * respectively. These methods return custom StreamObservers which are used to observe and respond to the streams.
+ *
+ */
+public class BlockStreamService implements GrpcService {
+
+    private final Logger logger = Logger.getLogger(getClass().getName());
```

Comment:
Our coding standard uses LOGGER vs. logger for this static field

Reply:
Fixed with the latest commit